mysql/data.go

```diff
@@ -126,9 +126,19 @@ func convScalar(spannerType ddl.Type, srcTypeName string, TimezoneOffset string,
 	}
 }
 
-func convBool(val string) (bool, error) {
+func convBool(conv *internal.Conv, val string) (bool, error) {
 	b, err := strconv.ParseBool(val)
 	if err != nil {
+		// MySQL allows non-boolean values in boolean columns.
+		// If we can't parse as boolean, try to re-parse as INT64
+		// and treat as true if value is non-zero.
+		// Note: if ParseBool fails, then val is probably a non-zero number.
+		i, err2 := convInt64(val)
+		if err2 == nil {
+			b = i != 0
```

Comment:
consider adding check that this value is in range -127 to 127 since we don't want to convert to bool otherwise

Reply:
Done.
Just to clarify: the allowed range for tinyint(1) is -128 to 127.